src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,25 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() == null
+                || user.getPassword() == null
+                || user.getLogin() == null
+                || user.getId() == null) {
+            throw new InvalidDataException();
+        }
+        if (user.getLogin().length() >= MIN_LENGTH
+                && user.getPassword().length() >= MIN_LENGTH
+                && user.getAge() >= MIN_AGE
```

Comment:
The condition `user.getLogin().length() >= MIN_LENGTH` seems to be incorrect, as the length check should be applied to the password, not the login.